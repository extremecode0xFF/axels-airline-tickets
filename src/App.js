import { Container, Row, Col } from 'react-bootstrap';
import {
  Filters,
  Header,
  ButtonTabs,
  ButtonMore,
  TicketList,
} from './components';
import { GlobalStyles } from './styled/GlobalStyles';

function App() {
  const list = [
    'Все',
    'Без пересадок',
    '1 пересадка',
    '2 пересадки',
    '3 пересадки',
  ];
  return (
    <>
      <GlobalStyles />
      <Container>
        <Header />
        <Row className="mx-auto" as="section">
          <Col md={3}>
            <Filters name="Количество пересадок" checkBoxList={list} />
          </Col>
          <Col>
            <ButtonTabs />
            <TicketList />
            <ButtonMore />
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default App;
