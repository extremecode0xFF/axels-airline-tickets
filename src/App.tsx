import React, { useState } from 'react';

import {
  Filters,
  Header,
  ButtonTabs,
  ButtonMore,
  TicketList,
} from './components';
import { Container, Row, Col } from 'react-bootstrap';
import { GlobalStyles } from './styled/GlobalStyles';

import { configFilterCheckbox, configFilterTabs } from './configs/params';
import { useTypedSelector } from './hooks/useTypedSelector';

function App() {
  const tickets = useTypedSelector((state) => state.tickets);
  const [visibleTickets, setVisibleTickets] = useState(5);

  return (
    <>
      <GlobalStyles />
      <Container>
        <Header />
        <Row className="mx-auto" as="section">
          <Col md={3}>
            <Filters
              name="Количество пересадок"
              config={configFilterCheckbox}
            />
          </Col>
          <Col>
            <ButtonTabs config={configFilterTabs} />
            <TicketList tickets={tickets} visibleTickets={visibleTickets} />
            <ButtonMore setShowMore={setVisibleTickets} />
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default App;
