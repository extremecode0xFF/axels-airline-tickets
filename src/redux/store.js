import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';

import reducer from './ducks/tickets';
import { watcherSaga } from './ducks/rootSaga';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(watcherSaga);

export default store;
