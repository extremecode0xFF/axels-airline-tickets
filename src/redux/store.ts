import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import { all } from 'redux-saga/effects';

import reducer from './ducks/tickets';
import {
  watcherGetTickets,
  watcherSetFilteredTickets,
  watcherSetSortedTickets,
} from './ducks/tickets';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

function* watcherSaga() {
  yield all([
    watcherGetTickets(),
    watcherSetFilteredTickets(),
    watcherSetSortedTickets(),
  ]);
}

sagaMiddleware.run(watcherSaga);

export type RootState = ReturnType<typeof store.getState>;

export default store;
