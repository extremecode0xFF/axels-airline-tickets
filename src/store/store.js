import { configureStore } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import { watcherSaga } from './sagas/rootSaga';
import reducer from './ducks/tickets';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(watcherSaga);

export { store };
