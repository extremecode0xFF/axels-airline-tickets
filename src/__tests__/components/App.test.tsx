import React from 'react';
import { render } from '@testing-library/react';
import store from '../../redux/store';

import App from '../../App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('App', () => {
  test('App make snapshot', () => {
    const app = (
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
    render(app);
    expect(app).toMatchSnapshot();
  });
});
