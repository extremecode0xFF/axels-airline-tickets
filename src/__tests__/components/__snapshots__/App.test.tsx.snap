// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App make snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <App />
  </BrowserRouter>
</Provider>
`;
